src/Http/Http.Extensions/src/HttpJsonServiceExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// Extension methods to configure JSON serialization behavior.
+/// </summary>
+public static class HttpJsonServiceExtensions
+{
+    /// <summary>
+    /// Configures options used for reading and writing JSON by route handlers.
+    /// </summary>
+    /// <remarks>
+    /// The options configured here will only affect JSON returned and processed
+    /// from route handlers, not controllers.
+    /// </remarks>
+    /// <param name="services">The <see cref="IServiceCollection" /> to configure options on.</param>
+    /// <param name="configureOptions">The <see cref="Action{JsonOptions}"/> to configure the
+    /// <see cref="JsonOptions"/>.</param>
+    /// <returns>The modified <see cref="IServiceCollection"/>.</returns>
+    public static IServiceCollection ConfigureHttpJsonOptions(this IServiceCollection services, Action<JsonOptions> configureOptions)
```

Comment:
We should review this name.

Reply:
Is this the first time we're adding a `ConfigureWhatever()` `IServiceCollection` extension method for an option type? I guess in most cases it wouldn't be necessary because you could have an overload that takes a `configureOptions` callback to `AddWhatever()` for most option types.

The only thing I see right now that's close to this is `ConfigureApplicationCookie()` and `ConfigureExternalCookie()` which use named options.

Reply:
Not exactly the same, but we added `ConfigureApiBehaviorOptions` to MVC to solve a similar discoverability issue: https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.mvccoremvcbuilderextensions.configureapibehavioroptions?view=aspnetcore-6.0#microsoft-extensions-dependencyinjection-mvccoremvcbuilderextensions-configureapibehavioroptions(microsoft-extensions-dependencyinjection-imvcbuilder-system-action((microsoft-aspnetcore-mvc-apibehavioroptions)))

Reply:
I updated the name to `ConfigureRouteHandlerJsonOptions` per our email conversation but open to other ideas.